### PR TITLE
Lbe 200 assignment grading once grading is done a mean median of the batch grade should be generated

### PR DESCRIPTION
**Pull request by gkav01**

### CODE REVIEW

File: src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java [hunk 1]
```diff
@@ -351,6 +351,62 @@ public List<AssignmentSubmitDTO> getGradesByProgramId(Long programId)
...
+		double sum = 0;
+		double mean;
+		
+		
+		List<AssignmentSubmit> assignmentSubmits = assignmentSubmitRepository.findByAssignment_Batch_BatchId(batchId);
+        if (assignmentSubmits.isEmpty()) {
+            throw new ResourceNotFoundException("Assignments with grades does not exist for Batch ID : "+batchId);
+        }	
+		
+		for (int i = 0; i < assignmentSubmits.size(); i++) {
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java:368]: Please use Java streams to do this operation, all these lines can be done in a single line

File: src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java [hunk 2]
```diff
@@ -351,6 +351,62 @@ public List<AssignmentSubmitDTO> getGradesByProgramId(Long programId)
...
+	public double getGradesMedianByBatchId(Integer batchId) {
+		AssignmentSubmit assignment;
+		 List<Integer> gradeList = new ArrayList<>();
+		
+		List<AssignmentSubmit> assignmentSubmits = assignmentSubmitRepository.findByAssignment_Batch_BatchId(batchId);
+       if (assignmentSubmits.isEmpty()) {
+           throw new ResourceNotFoundException("Assignments with grades does not exist for Batch ID : "+batchId);
+       }	
+		
+		for (int i = 0; i < assignmentSubmits.size(); i++) {
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java:390]: Same as above. Try using streams

File: src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java [hunk 3]
```diff
@@ -351,6 +351,62 @@ public List<AssignmentSubmitDTO> getGradesByProgramId(Long programId)
...
+			assignment = (AssignmentSubmit)assignmentSubmits.get(i);
+			gradeList.add(assignment.getGrade());
+		}
+		System.out.println("unsorted list= "+gradeList);
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java]: Change all System.out.println to log.debug messages

File: src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java [hunk 4]
```diff
@@ -351,6 +351,62 @@ public List<AssignmentSubmitDTO> getGradesByProgramId(Long programId)
...
+		Collections.sort(gradeList);
+		int length = gradeList.size();
+		System.out.println("sorted list= "+gradeList);
+		System.out.println("length= "+length);
+		 int middle = length/2;
+	    if (length%2 == 1) { 
+	    	System.out.println("median= "+(gradeList.get(middle)));
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java]: This line can be written only once, add a logger and use log.debug

File: src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java [hunk 5]
```diff
@@ -351,6 +351,62 @@ public List<AssignmentSubmitDTO> getGradesByProgramId(Long programId)
...
+		for (int i = 0; i < assignmentSubmits.size(); i++) {
+			assignment = (AssignmentSubmit)assignmentSubmits.get(i);
+			gradeList.add(assignment.getGrade());
+		}
+		System.out.println("unsorted list= "+gradeList);
+		Collections.sort(gradeList);
+		int length = gradeList.size();
+		System.out.println("sorted list= "+gradeList);
+		System.out.println("length= "+length);
+		 int middle = length/2;
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/services/AssignmentSubmitService.java]: Pls handle any required corner conditions-
ex: when only one entry is present

**Review by gkav01 — COMMENTED** [review]
Please write Junit cases for Mean and median. 
Make the quested changes, test and merge the code.
